core/src/main/clojure/xtdb/xtql.clj

```diff
@@ -319,33 +349,67 @@
             (plan-query (.query pipeline))
             (.tails pipeline))))
 
+(defn- wrap-scalar-subquery [plan {:keys [placeholder subquery args] :as x}]
+  (let [{:keys [ra-plan provided-vars]} subquery
+        projected-subquery [:project [{placeholder (first provided-vars)}]
+                            ra-plan]]
+    (when-not (= (count provided-vars) 1)
+      (throw (err/illegal-arg
+              :xtql/invalid-scalar-subquery
+              {:subquery subquery :provided-vars provided-vars
+               ::err/message "Scalar subquery must only return a single column"})))
+    (if (seq args)
+      [:apply
+       :single-join
+       (set/map-invert (args->apply-params args)) ;;TODO push this invert into args->apply-params?
+       plan
+       projected-subquery]
+      [:single-join [true]
+       plan
+       projected-subquery])))
+
+(defn- wrap-scalar-subqueries [plan subqueries]
+  (reduce
+   (fn [plan sq]
+     (wrap-scalar-subquery plan sq))
+   plan
+  subqueries))
+
+(defn- wrap-project [plan projection]
+  [:project projection plan])
+
 (extend-protocol PlanQueryTail
   Query$Where
   (plan-query-tail [this {:keys [ra-plan provided-vars]}]
-    {:ra-plan (wrap-select
-               ra-plan
-               (map
-                #(do
-                   (required-vars-available? % provided-vars)
-                   (plan-expr %))
-                (.preds this)))
-     :provided-vars provided-vars})
+
+    (doseq [pred (.preds this)]
+      (required-vars-available? pred provided-vars))
+
+    (let [planned-where-exprs (plan-where this)]
+      {:ra-plan (-> ra-plan
+                    (wrap-scalar-subqueries (mapcat :subqueries planned-where-exprs))
+                    (wrap-select (map :expr planned-where-exprs))
+                    (wrap-project (vec provided-vars)))
+       :provided-vars provided-vars}))
 
   Query$WithCols
   (plan-query-tail [this {:keys [ra-plan provided-vars]}]
+    ;;TODO Needs to check if projected col already exists otherwise bad things happen.
+    ;;Should we assume error here? or should we allow the user to project over the existing col.
```

Comment:
@mbutlerw should allow the user to project over the existing col in the query-tail operator